rpcserver.go

```diff
@@ -1913,6 +1913,33 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 			err)
 	}
 
+	var channelType *lnwire.ChannelType
+	switch in.CommitmentType {
+	case lnrpc.CommitmentType_UNKNOWN_COMMITMENT_TYPE:
+		break
```

Comment:
Shouldn't this instead set the current "default" type, so we handle the case earlier vs needing to reoncile w/ a nil/unknown chan type by the time we get all the way into the funding manager.

Reply:
The current negotiation flow will fail if we try explicit negotiation and the remote peer doesn't advertise the feature bit, so setting a default here would prevent opening channels to any node that doesn't. If we allow the negotiation to fall back to implicit, then we'd need a distinct parameter to determine whether the explicit type being negotiated is a default set by us or set by the RPC user. In the latter case, we wouldn't want to fall back to implicit as that would go against what the user requested.